### PR TITLE
Issues fixes

### PR DESCRIPTION
**Pull request by Brainenjii**

### CODE REVIEW

File: src/containers/Activity/Activity.js [hunk 1]
```diff
@@ -57,7 +57,9 @@ export class Activity extends React.PureComponent {
       );
     }
   }
-  componentWillReceiveProps(nextProps) {
+
+  // Renamed via https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops
+  UNSAFE_componentWillReceiveProps(nextProps) {
```
Comment (FeynmanDNA) [inline src/containers/Activity/Activity.js]: @Brainenjii should we change this to `componentDidUpdate()` and `getDerivedStateFromProps()`?

File: src/containers/Activity/Activity.js [hunk 2]
```diff
@@ -50,11 +57,9 @@ export class Activity extends React.PureComponent {
       );
     }
   }
-  componentWillReceiveProps(nextProps){
-    if(
-      !isEqual(nextProps.pathProblem, this.props.pathProblem)
-    ){
-    this.onProblemChange({});
+  componentWillReceiveProps(nextProps) {
```
Comment (FeynmanDNA) [inline src/containers/Activity/Activity.js]: @Brainenjii do you think `componentWillReceiveProps()` in our code base should all be replaced?

File: src/containers/HomeV2/SampleCarousel.jsx [hunk 3]
```diff
@@ -59,71 +58,66 @@ class SampleCarousel extends React.PureComponent {
         break;
       default:
         itemPerSlide = 4;
-    };
+    }
 
     return (
       <Carousel
-        slidesToShow={itemPerSlide}
```
Comment (FeynmanDNA) [inline src/containers/HomeV2/SampleCarousel.jsx]: @Brainenjii do we need `slidesToShow` for our GUI? Without it the breakpoint of screenwidth will not work.

File: src/containers/HomeV2/HomeV2.js [hunk 4]
```diff
@@ -69,40 +76,48 @@ export class HomeV2 extends React.Component {
...
             }
 
-            return recommendedData.length>0 ? (
+            return recommendedData.length > 0 ? (
               <RecommendationListCard
+                dummyData={recommendedData}
                 key={recommendationKey}
-                dummyData={ recommendedData }
                 RecomType={
                   recommendationKey === "youtube" ? "youtube" : "python"
```
Comment (FeynmanDNA) [inline src/containers/HomeV2/HomeV2.js]: @Brainenjii we might want to expand the recommendationKey to more than just `python` and `youtube`.

File: src/containers/HomeV2/RecommendationListCard.jsx [hunk 5]
```diff
@@ -66,41 +66,38 @@ class RecommendationListCard extends React.PureComponent {
...
                   feel free to refactor it to a proper solution */}
               <CardHeader
                 avatar={
-                  <Avatar aria-label="Recommendation"
-                    style={isCodeCombat
-                      ? styles.avatarCC
-                      : styles.avatarPy}
+                  <Avatar
+                    aria-label="Recommendation"
+                    style={isCodeCombat ? styles.avatarCC : styles.avatarPy}
```
Comment (FeynmanDNA) [inline src/containers/HomeV2/RecommendationListCard.jsx]: @Brainenjii avatars need to be specific for different activity types, Jest, Game etc

File: src/containers/HomeV2/RecommendationListCard.jsx [hunk 6]
```diff
@@ -66,41 +66,38 @@ class RecommendationListCard extends React.PureComponent {
...
                 }
                 subheader="Recommended for you"
-                title={title|| "Jupyter Notebook Activities"}
+                title={title || "Jupyter Notebook Activities"}
               />
               <CardContent>
                 <SampleCarousel
                   dataList={dummyData}
-                  youtubeRecom={false}
                   isCodeCombat={isCodeCombat}
```
Comment (FeynmanDNA) [inline src/containers/HomeV2/RecommendationListCard.jsx]: @Brainenjii `isCodeCombat` was inserted as temporary fix for the GUI to render a row for CodeCombat type activities. There should be some refactoring to expand the detection of activity types.

**Comment by FeynmanDNA** [comment]
https://github.com/NUS-ALSET/achievements/issues/297#issuecomment-415249451

File: src/containers/Activity/Activity.js (hunk 2 above)
Comment (Brainenjii) [reply src/containers/Activity/Activity.js]: Yeah, I think we should consider avoiding using deprecated (and maybe also unsafe) features

File: src/containers/HomeV2/SampleCarousel.jsx (hunk 3 above)
Comment (Brainenjii) [reply src/containers/HomeV2/SampleCarousel.jsx]: I think we need it. Without this prop carousel always shows only 1 item.

**Comment by Brainenjii** [comment]
@FeynmanDNA updated code of HomeV2. Could you review your notes?

**Comment by FeynmanDNA** [comment]
@Brainenjii the update looks awesome!

I think the new `solvesPySkills` and `unSolvedPySkills` are work in progress from [commit](6287d0) by @walkwel

**Pull request merged**